lib/omnibus/health_check.rb

```diff
@@ -135,6 +135,8 @@ class HealthCheck
       /libSystem\.B\.dylib/,
       # Symlink of the previous one
       /libgcc_s\.1\.dylib/,
+      /libz\.1\.dylib/,
+      /libdatadog-agent-rtloader\.1\.dylib/,
```

Comment:
Context is missing for this.

1. It might be fine for `libz` to be whitelisted; this just means we're fine with _potentially_ relying on the version shipped with macOS, though I think it should be a conscious decision and avoid shipping it if we're planning on relying on the system version. At any rate, it's not clear how this is related to the task at hand.
2. For libdatadog-agent-rtloader, I doubt this is a fine thing to whitelist; but at any rate I really think this should not be whitelisted inside omnibus-ruby, if anything it should live close to where this is actually built.

Reply:
After investigation we realized that the health check isn't interpreting `@loader_path`when inside of `@rpath` causing the issue. Fix of the health check in progress + removal both of these lib inside of the whitelist